opm/input/eclipse/Schedule/MSW/SegmentMatcher.hpp

```diff
@@ -0,0 +1,303 @@
+        }
+
+        /// Retrieve list of (MS) well names covered by this result set.
+        ///
+        /// \return List MS well names covered by this result set.
+        std::vector<std::string_view> wells() const;
+
+        /// Retrive result set's segments for single MS well.
+        ///
+        /// \param[in] well Named MS well.  Should usually be one of the
+        ///    items in the return value from \code wells() \endcode.
+        ///
+        /// \return range of \c well's segments matching the input request.
+        ///    Empty unless \p well is one of the return values from \code
+        ///    wells() \endcode.
+        WellSegmentRange segments(std::string_view well) const;
+
+        friend class SegmentMatcher;
+
+    private:
+        /// List of MS wells covered by this result set.
+        std::vector<std::string> wells_{};
+
+        /// Sorted index into \c wells_ for faster lookup.
+        ///
+        /// User must call \code sort() \endcode to establish index.
+        std::vector<std::vector<std::string>::size_type> sortedWells_{};
+
+        /// CSR start pointers for MS wells' segments.
+        std::vector<std::vector<int>::size_type> segmentStart_{};
+
+        /// All segments covered by this result set.  Structured by \c
+        /// segmentStart_.
+        std::vector<int> segments_{};
+
+        /// Sort result set for quicker lookup from well names.
+        ///
+        /// For use by SegmentMatcher only.
+        void sort();
+
+        /// Add non-empty range of segments for single MS well to result set.
+        ///
+        /// For use by SegmentMatcher only.
+        ///
+        /// \param[in] well Name of MS well.
+        ///
+        /// \param[in] segments List of segment numbers matching input
+        ///    request for \p well.
+        void addWellSegments(const std::string&      well,
+                             const std::vector<int>& segments);
+    };
+
+    /// Description of Particular Segment Set
+    ///
+    /// User specified.
+    class SetDescriptor
+    {
+    public:
+        /// Default Constructor
+        SetDescriptor() = default;
+
+        /// Assign request's segment number.
+        ///
+        /// Non-positive matches all segments.
+        ///
+        /// \param[in] segNum Requests's segment number.
+        ///
+        /// \return \code *this \endcode.
+        SetDescriptor& segmentNumber(const int segNum);
+
+        /// Assign request's segment number.
+        ///
+        /// String version.  Supports both quoted and unquoted strings.
+        /// Wildcard ('*') and string representation of a negative number
+        /// (e.g., '-1'), match all segments.
+        ///
+        /// \param[in] segNum Requests's segment number.  Must be a text
+        ///    representation of an integer or one of the recognized options
+        ///    for matching all segments.  Throws exception otherwise.
+        ///
```

Comment:
I think the documentation would be easier to find if  it would move to the SegmentMatcher class and here a link would be added to there. Doesn't "0", 0 also match all segments?

Reply:
If there ever surfaces a bug in this area that would help (save time), though.